cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -81,18 +84,23 @@
   private final Service internalService;
   private final DatasetFramework datasetFramework;
   private final TimeSchedulerService timeSchedulerService;
+  private final AdminEventPublisher adminEventPublisher;
+  private final MultiThreadMessagingContext messagingContext;
 
   @Inject
   CoreSchedulerService(TransactionSystemClient txClient, DatasetFramework datasetFramework,
                        TimeSchedulerService timeSchedulerService,
                        ScheduleNotificationSubscriberService scheduleNotificationSubscriberService,
-                       ConstraintCheckerService constraintCheckerService) {
+                       ConstraintCheckerService constraintCheckerService,
+                       AdminEventPublisher adminEventPublisher,
+                       MessagingService messagingService) {
     this.startedLatch = new CountDownLatch(1);
     this.datasetFramework = datasetFramework;
+    this.messagingContext = new MultiThreadMessagingContext(messagingService);
     DynamicDatasetCache datasetCache =
       new MultiThreadDatasetCache(new SystemDatasetInstantiator(datasetFramework),
                                   txClient, Schedulers.STORE_DATASET_ID.getParent(),
-                                  Collections.emptyMap(), null, null);
+                                  Collections.emptyMap(), null, null, messagingContext);
```

Comment:
I guess this is why messaging context is passed into the event publisher, because it's shared across multiple classes? You can just modify the constructor to pass in the cConf and messaging context without injection, just like the the MultiThreadDatasetCache is doing.

Reply:
Fixed